File: parser/src/main/java/com/github/nsc/de/shake/parser/node/logical/LogicalNode.java
package com.github.nsc.de.shake.parser.node.logical;

import com.github.nsc.de.shake.parser.node.ValuedNode;
import com.github.nsc.de.shake.util.characterinput.position.PositionMap;

public abstract class LogicalNode extends ValuedNode {

    public LogicalNode(PositionMap map) {
        super(map);
    }

    abstract String getOperator();

}

File: parser/src/main/kotlin/com/github/nsc/de/shake/parser/node/logical/LogicalNode.kt
package com.github.nsc.de.shake.parser.node.logical

import com.github.nsc.de.shake.parser.node.ValuedNode
import com.github.nsc.de.shake.parser.node.logical.LogicalConcatenationNode
import com.github.nsc.de.shake.parser.node.logical.LogicalCompareNode
import com.github.nsc.de.shake.parser.node.logical.LogicalNode
import com.github.nsc.de.shake.util.characterinput.position.PositionMap

abstract class LogicalNode(map: PositionMap) : ValuedNode(map) {
    abstract val operator: String?
}